plantcv/annotate/classes.py

```diff
@@ -158,3 +162,365 @@ class label, by default "default"
+                                debug_img_duplicates = np.where(labeled_mask_all == mask_pixel_value,
+                                                                (255), debug_img_duplicates)
+                                # Fill in the duplicate object in the labeled mask, replace with pixel annotations
+                                final_mask = np.where(labeled_mask_all == mask_pixel_value, (0), final_mask)
+                                added_obj_labels.append(mask_pixel_value)
+                                for dup_coord in associated_coords:
+                                    # Draw each pixel in the final mask
+                                    final_mask[dup_coord] = object_id_count
+                                    analysis_labels.append(names)
+                                    # Add a thicker pixel where unresolved annotation to the debug img
+                                    cv2.circle(debug_img, (dup_coord[1], dup_coord[0]), radius=params.line_thickness,
+                                               color=(object_id_count), thickness=-1)
+                                    # Add debug label annotations later
+                                    debug_labels, debug_coords = _add_debug_id(debug_labels, debug_coords,
+                                                                               object_id_count, (dup_coord[1], dup_coord[0]))
+                                    # Increment object count up so each pixel drawn in labeled mask is unique
+                                    object_id_count += 1
+                            if len(re) > 1:
+                                # More than one class label associated with a given object
+                                splitup = []
+                                # Split on "_" in case something has already been combined
+                                for lbls in coord_labels:
+                                    list_lbl = []
+                                    for lbl in lbls:
+                                        list_lbl.append(lbl.split("_"))
+                                    splitup.append(np.concatenate(list_lbl))
+                                # Flatten list of labels
+                                flat = np.concatenate(splitup)
+                                # Grab each unique label from the list
+                                unique_lbls, lbl_counts = np.unique(flat, return_counts=True)
+                                # Is there duplication within each class label for the given object?
+                                if np.all(lbl_counts == 1):
+                                    # If no, Concat with "_" delimiter
+                                    concat_lbl = "_".join(list(unique_lbls))
+                                    if params.verbose:
+                                        print(f"labels getting concatenated to '{concat_lbl}' at {first_coord}")
```

Comment:
Remove if and use `pcv.warn` instead of `print`. (side note: we should update `warn` to respect `pcv.params.verbose`